lib/rubocop/cop/rspec/expect_in_hook.rb

```diff
@@ -37,6 +47,14 @@ def on_block(node)
 
         private
 
+        def nested_in_shared_group?(node)
+          node.each_ancestor(:block).any?(&method(:shared_group?))
```

Comment:
I can think of the following, where expect in hook is not directly nested in a shared group:
```ruby
shared_example 'some shared examles' do
  describe '#some_method' do
    context 'in some case' do
      after do
        expect_any_instance_of(Something).to receive(:foo) # NASTY!
      end

      it 'does that' do
        something.perform
      end
    end
  end
end
```

Reply:
Good catch @pirj! I'll add an extra spec to cover both and make it explicit that it works in all the levels.


Also, it's also not being actually covered, looks like the implementation needs some fixes too :D